lib/grandpa/grandpa.go

```diff
@@ -529,124 +518,59 @@ func (s *Service) playGrandpaRound() error {
 	if err != nil {
 		return err
 	}
-
 	s.precommits.Store(s.publicKeyBytes(), spc)
-	logger.Debugf("sending pre-commit message %s...", pc)
 
-	// continue to send precommit messages until round is done
-	go s.sendVoteMessage(precommit, pcm, roundComplete)
-
-	if err = s.attemptToFinalize(); err != nil {
-		logger.Errorf("failed to finalise: %s", err)
-		return err
-	}
+	go s.sendPrecommitMessage(pcm, finalizableCh)
+	// waits until round is finalizable
+	<-finalizableCh
 
 	logger.Debugf("round completed in %s", time.Since(start))
 	return nil
 }
 
-func (s *Service) sendVoteMessage(stage Subround, msg *VoteMessage, roundComplete <-chan struct{}) {
+func (s *Service) sendPrecommitMessage(vm *VoteMessage, isFinalizable <-chan struct{}) {
+	logger.Debugf("sending pre-commit message %s...", vm.Message)
+
 	ticker := time.NewTicker(s.interval * 4)
 	defer ticker.Stop()
 
 	// Though this looks like we are sending messages multiple times,
 	// caching would make sure that they are being sent only once.
 	for {
-		if s.paused.Load().(bool) {
+		select {
+		case <-ticker.C:
+		case <-isFinalizable:
 			return
 		}
 
```

Comment:
do we need the for loop here still? comment is a bit sus too (i know it works but still lol)